qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/texture/presenter.py

```diff
@@ -0,0 +1,351 @@
+            self.model.set_readout_col(self.view.get_readout_column())
+            self.model.set_out_ws_and_grouping(wss, params)
+            ax_transform, ax_labels = self._get_ax_data()
+            self.model.set_ax_trans(ax_transform)
+            self.model.set_ax_labels(ax_labels)
+            self.model.set_plot_exp(self._get_setting("plot_exp_pf", bool))
+            self.model.set_contour_kernel(float(self._get_setting("contour_kernel", str)))
+
+            # default scattering pos for now
+            self.model.set_scat_vol_pos((0.0, 0.0, 0.0))
+
+            # get the threshold values from the view
+            has_chi2_col, has_x0_col = False, False
+            if self.all_wss_have_params() and self.at_least_one_param_assigned():
+                has_chi2_col, has_x0_col = self.model.check_param_ws_for_columns(params)
+            chi2_thresh = self._get_setting("cost_func_thresh") if has_chi2_col else 0.0
+            peak_thresh = self._get_setting("peak_pos_thresh") if has_x0_col else 0.0
+
+            self.model.set_chi2_thresh(chi2_thresh)
+            self.model.set_peak_thresh(peak_thresh)
+
+            self.set_worker(
+                AsyncTask(
+                    self.calc_pf,
+                    (
+                        wss,
+                        params,
+                    ),
+                    error_cb=self._on_worker_error,
+                    finished_cb=self._on_worker_success,
+                )
+            )
+            self.worker = self.get_worker()
+            self.worker.start()
+
+    def _get_ax_data(self):
+        return output_settings.get_texture_axes_transform()
+
+    def set_worker(self, worker):
+        self.worker = worker
+
+    def get_worker(self):
+        return self.worker
+
+    def calc_pf(
+        self,
+        wss,
+        params,
+    ):
+        root_dir = output_settings.get_output_path()
+        save_dirs = self.model.get_save_dirs(root_dir, "PoleFigureTables", self.model.get_rb_num(), self.model.get_grouping())
+        self.model.make_pole_figure_tables(
+            wss,
+            params,
+            self.model.get_out_ws(),
+            self.model.get_hkl(),
+            self.model.get_inc_scatt(),
+            self.model.get_scat_vol_pos(),
+            self.model.get_chi2_thresh(),
+            self.model.get_peak_thresh(),
+            save_dirs,
+            self.model.get_ax_trans(),
+            self.model.get_readout_col(),
+        )
+        self.plot_pf(save_dirs)
+
+    def _on_worker_success(self):
+        self.correction_notifier.notify_subscribers("Pole Figure Created")
+
+    def _on_worker_error(self, error_info):
+        logger.error(str(error_info))
+
+    def plot_pf(self, save_dirs):
+        # Get figure and canvas from view
+        fig, canvas = self.view.get_plot_axis()
+
+        # Clear existing figure
+        fig.clf()
+
+        self.model.plot_pole_figure(
+            self.model.out_ws,
```

Comment:
Same here - is the difficulty that the model function is designed to be used by the scripting interface/model class which doesn't have these getters? In which case I would just override the method in the GUI inherited class (if that makes sense?).